Sources/SPMBuildCore/PluginInvocation.swift

```diff
@@ -502,29 +502,31 @@ public extension PluginTarget {
     }
 
     /// The set of tools that are accessible to this plugin.
-    func accessibleTools(fileSystem: FileSystem, environment: BuildEnvironment, for hostTriple: Triple) -> Set<PluginAccessibleTool> {
-        return Set(self.dependencies(satisfying: environment).flatMap { dependency -> [PluginAccessibleTool] in
+    func accessibleTools(packageGraph: PackageGraph, fileSystem: FileSystem, environment: BuildEnvironment, for hostTriple: Triple) throws -> Set<PluginAccessibleTool> {
+        return try Set(self.dependencies(satisfying: environment).flatMap { dependency -> [PluginAccessibleTool] in
+            let executableOrBinaryTarget: Target
             switch dependency {
             case .target(let target, _):
-                // For a binary target we create a `vendedTool`.
-                if let target = target as? BinaryTarget {
-                    // TODO: Memoize this result for the host triple
-                    guard let execInfos = try? target.parseArtifactArchives(for: hostTriple, fileSystem: fileSystem) else {
-                        // TODO: Deal better with errors in parsing the artifacts
-                        return []
-                    }
-                    return execInfos.map{ .vendedTool(name: $0.name, path: $0.executablePath) }
-                }
-                // For an executable target we create a `builtTool`.
-                else if target.type == .executable {
-                    // TODO: How do we determine what the executable name will be for the host platform?
```

Comment:
Note: I deleted this TODO because supporting the case where plugins work when the target platform isn't the host requires significant work in SwiftPM's build system.